### DISCUSSION TITLE
Can shell routes be deactivated?

### DISCUSSION THREAD

**Discussion by munzili** [AvaloniaInside/Shell#52] [Q&A]
I have a ShellView with 5 buttons at the bottom, and no sidebar is used. The main target is an Android UI.

On the starting page, the user needs to select some options. This data is required to display the other pages within the ShellView correctly.

Is it possible to use something like IsEnabled with a binding on the route, or hide the buttons?

By default, 2 Buttons/routes should be enabled and the other be deactivated. Just by make some changes in a combobox, the pages should be enabled. 

Best regards

**Comment by OmidID** [comment]
Hello @munzili 

Your problem is little blurry for me, I think this is related to your app logic not Shell or navigation system.

If you need to enable or disable your button base on some condition, you need to do it in your app logic not disable or enable routes. basically routes in app should be affected and this is the best practices. 

Maybe you need to add or remove dynamically routes and then you need to use Registrar to do it. You can find Registrar in Navigator or from DI you can find `INavigationRegistrar`

It help if you send screenshots and more clarify  

Thanks
